validphys2/src/validphys/n3fit_data.py

```diff
@@ -359,10 +359,20 @@ def pseudodata_table(groups_replicas_indexed_make_replica, replicas):
     `fitting::savepseudodata` is `true` (as per the default setting) and
     replicas are fitted one at a time. The table can be found in the replica
     folder i.e. <fit dir>/nnfit/replica_*/
-
     """
-    # Concatenate over replicas
-    df = pd.concat(groups_replicas_indexed_make_replica)
+    # groups_replicas_indexed_make_replica is collected over both replicas and dataset_input groups
+    # to correctly put this into a single dataframe, we first need to know the number of
+    # dataset_input groups there are for each replica
+    groups_per_replica = int(len(groups_replicas_indexed_make_replica) / len(replicas))
```

Comment:
```suggestion
    groups_per_replica = len(groups_replicas_indexed_make_replica) // len(replicas)
```

Reply:
It's used as a list index so it has to be `int`

Reply:
The output of `//` should be int.

Reply:
No, you are doing `len` vs `len` it should take the same type as the operands.

Reply:
Ah! You're right!

Reply:
If it is because of taste I'm ok with the `int()`.

Reply:
I also didn't realise it retained the original types, so it wasn't just taste. I already accepted this now